sdk/src/main/java/com/amplitude/experiment/DefaultExperimentClient.kt

```diff
@@ -105,20 +111,44 @@ internal class DefaultExperimentClient internal constructor(
 
     override fun variant(key: String, fallback: Variant?): Variant {
         val variantAndSource = resolveVariantAndSource(key, fallback)
-        val variant = variantAndSource.variant
-        val source = variantAndSource.source
         if (config.automaticExposureTracking) {
-            exposureInternal(key, variant, source)
+            exposureInternal(key, variantAndSource)
         }
-        return variant
+        return variantAndSource.variant
     }
 
     override fun exposure(key: String) {
-        val variantAndSource = resolveVariantAndSource(key, null)
-        exposureInternal(key, variantAndSource.variant, variantAndSource.source)
+        val variantAndSource = resolveVariantAndSource(key)
+        exposureInternal(key, variantAndSource)
+    }
+
+    private fun exposureInternal(key: String, variantAndSource: VariantAndSource) {
+        legacyExposureInternal(key, variantAndSource.variant, variantAndSource.source)
+
+        // Do not track exposure for fallback variants that are not associated with a default variant.
+        val fallback = isFallback(variantAndSource.source)
+        if (fallback && (variantAndSource.hasDefaultVariant == null || !variantAndSource.hasDefaultVariant)) {
+            return
+        }
+
+        val experimentKey = variantAndSource.variant.expKey
+        var variant: String? = null
+        val metadata = variantAndSource.variant.metadata
+        if (!fallback && (metadata?.get("default") == null)) {
```

Comment:
This logic is not quite right. In JavaScript the `!metadata?.default` will return true if:
1. metadata is null/undefined
2. default is null/undefined
3. default is falsey (false, 0, etc.)

To encapsulate all this logic, on iOS I wrote an extension function, `isDefaultVariant()` on `Variant`:

```
    func isDefaultVariant() -> Bool {
        if let isDefault = metadata?["default"] as? Bool {
            return isDefault
        }
        return false
    }
``` 

This is in swift, so let me know if it doesn't make sense.

You should use this here, and everywhere else that you're checking if the variant is default.